rfp-proposals/rfp-proposal-devnet-scripts.md

```diff
@@ -0,0 +1,148 @@
+# RFP Proposal: `Devnet scripts`
+
+**Name of Project: Devnet automation with Kubernetes**
+
+**Link to RFP:** https://github.com/filecoin-project/devgrants/blob/master/rfps/new-wave-4-rfps.md#devnet-scripts
+
+**Proposal Category:** `devtools-libraries`
+
+**Proposer:** `mike1729`
+
+**Do you agree to open source all work you do on behalf of this RFP and dual-license under MIT and APACHE2 licenses?:** "Yes"
+
+# Project Description
+
+The motivation behind this proposal is to make it easier to set up, customize, and scale a shared Filecoin Devnet.
+
+The solution is going to utilize Kubernetes scripts to automate setting up the network consisting of bootstrap nodes, faucet, and the ability to add additional miners from a prespecified whitelist.
+
+
+## Value
+
+This project is an added value for the Filecoin ecosystem as it lowers the complexity of spinning up a development environment that more closely resembles the actual mainnet. It enables faster development, easier testing, and shared collaboration for parties interested in testing in a real environment without experiencing issues with setting up a one.
+
+## Deliverables
+
+Kubernetes based automation scripts for running devnet environments.
+
+
+## Development Roadmap
+
+### Milestone 1 - Project and Preparation
+
+
+Goals:
+
+1.  Technical documentation and schematics for Kubernetes infrastructure.
+
+	 *  1 team member
+	 *  1 week
+	 * 1000$
+
+Time: 1 week
+Cost: 1000 $
+
+### Milestone 2 - Run Kubernetes Infrastructure
+
+Goals:
+
+1.  YAML configuration files for: bootstrap, faucet, miner and explorer nodes.
+
+	* 2 team members
+	* 1 week
+	* 4000$  
+
+2.  Underlying K8s infrastructure: deployments, Kubelet monitoring registration, namespaces.
+
+	* 2 team members
+	* 1 week
+	* 4000$
+
+Time: 2 week
+Cost: 8000 $
+
+### Milestone 3 - Make Application Run on Kubernetes
+
+Goals:
+
+1.  New dockerfiles or fitting already existing ones.
+
+	* 1 team member
+	* 1 week
+	* 1000$
+
+2.  Security improvements: whitelisting, role based access control.
+
+	* 1 team member
+	* 1 week
+	* 2000$
+
+Time: 1 week
+Cost: 3000$
+
+### Milestone 4 - Samples to Run Project on AWS
+
+Goals:
+
+1.  Terraform modules for setting sample infrastructure on AWS.
+
+	* 1 team member
+	* 1 week
+	* 2000$
+
+Time: 1 week
+Cost: 2000 $
+
+### Milestone 5 - Configure & Automate For Easier Use
+
+Goals:
+
+1.  Developing configuration and automation scripts in Golang:
```

Comment:
Hey @mike1729,

Thanks for your proposal! Can you explain why your team doesn't want to use a k8 Helm Chart here?

Reply:
It's not that we don't want to use Helm. Pre v3, Helm was not really working all that great so we tend to ask people about it just to be certain of the direction. It's still something that we use extensively, especially at scale, when setting multiple environments quickly (based on the same template) is key. We'll be happy to implement Helm for this project as well! 

What's worth keeping in the back of all our minds is that sometimes, it tends to be a bit too much for some teams. Especially if they're not acquainted with Helm or Kubernetes.